src/main/kotlin/org/jitsi/jibri/util/extensions/SchedulerExecutorServiceExts.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright @ 2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.jitsi.jibri.util.extensions
+
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+/**
+ * A version of [ScheduledExecutorService.scheduleAtFixedRate] that takes
+ * a lambda instead of requiring a [Runnable] (and moves it to the last
+ * argument)
+ */
+fun ScheduledExecutorService.scheduleAtFixedRate(
+    delay: Long = 0,
```

Comment:
Since we a redeviating from Java's argument order anyway, how about moving `delay` to the third place (since it has a default value)? We can then removes the named arguments from it's usage:
`scheduleAtFixedRate(period = 15, unit = TimeUnit.SECONDS) `

Reply:
this actually exists to move the function to the last argument so the
```
scheduleAtFixedRate(delay, unit) {
 // func
}
```
syntax can be used, if the function isn't the last one we can't use the {} syntax :/

Reply:
My suggestion is to move `delay` (which has a default value) after `period` and `unit`:
```
fun ScheduledExecutorService.scheduleAtFixedRate(
    period: Long,
    unit: TimeUnit,
    delay: Long = 0,
    action: () -> Unit
):
```

This allows us to call with `scheduleAtFixedRate(15, TimeUnit.SECONDS)` instead of having to specify the argument names: `scheduleAtFixedRate(period = 15, unit = TimeUnit.SECONDS)`